### PR TITLE
fix badmatch in case of non-existent bucket ( #385 )

### PR DESCRIPTION
**Pull request by kuenishi**

### CODE REVIEW

**Comment by shino** [comment]
tiny typo, s/bad much/bad match/

File: src/riak_cs_wm_utils.erl [hunk 1]
```diff
@@ -414,63 +414,74 @@ bucket_access_authorize_helper(AccessType, Deletable,
...
-            %% undefined here
-            AccessRD = riak_cs_access_logger:set_user(User, RD),
-            Access = PolicyMod:reqdata_to_access(RD, AccessType,
-                                                 User?RCS_USER.canonical_id),
-            case PolicyMod:eval(Access, Policy) of
-                true ->      {false, AccessRD, PermCtx};
-                undefined -> {false, AccessRD, PermCtx};
-                false ->     riak_cs_wm_utils:deny_access(AccessRD, Ctx)
-            end;
+        {error, Reason} when is_atom(Reason) ->
```
Comment (shino) [inline src/riak_cs_wm_utils.erl:421]: I don't understand why this is_atom(Reason) is sperated from L.424.
This case clause responds "503 Service Unavailable".
On the other hand, {error, _} clause at L.424 responds "404 Object Not Found".

**Comment by kuenishi** [comment]
Error atoms are defined in riak_cs_s3_response and get_bucket_acl_policy may or may not returns such atoms.

I thought returning 'false'  in authorize/2 would cause 403 not authorized, doesn't it?

File: src/riak_cs_wm_utils.erl (hunk 1 above)
Comment (kuenishi) [inline src/riak_cs_wm_utils.erl:421]: ok, this case should not be matched, thus I removed this match line to return 500, in case of unexpected errors.

**Comment by shino** [comment]
Authorization is determined in `{ok, {BucketAcl, Policy}}` clause.
These clauses are executed, for example, PB/networking error, so 403 is misleading.

**Comment by kuenishi** [comment]
yep, I removed that patter match to let it die and return 500.

**Comment by jrwest** [comment]
ran the riak_tests against this branch. all previously failing tests passed

**Pull request merged**